File: src/baekJoon/BJ8958.java
package baekJoon;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import static java.lang.Integer.parseInt;

public class BJ8958 {

    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int count = parseInt(bufferedReader.readLine());
        int[] point = new int[count];

        for (int i = 0; i < count; i++) {
            String[] OX = bufferedReader.readLine().split("");
            int repeat = 0;

            for (String ox : OX) {

                if (ox.equals("O")) {
                    repeat++;
                    point[i] += repeat;
                } else if (ox.equals("X")) {
                    repeat = 0;
                }
            }
        }

        for (int p : point) {
            System.out.println(p);
        }

    }
}

File: src/baekJoon/BJ8958.kt
package baekJoon

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()

    var n = br.readLine().toInt()

    while (n-- > 0) {
        val oxArr = br.readLine().toCharArray()

        var acc = 0;
        var total = 0;
        for (ox in oxArr) {
            if (ox == 'O') {
                total += ++acc;
            } else {
                acc = 0;
            }
        }

        sb.append(total).append("\n")
    }

    println(sb)
}